include/aws/common/math.msvc.inl

```diff
@@ -44,21 +44,36 @@ AWS_STATIC_IMPL int aws_mul_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
  * Multiplies a * b. If the result overflows, returns 2^32 - 1.
  */
 AWS_STATIC_IMPL uint32_t aws_mul_u32_saturating(uint32_t a, uint32_t b) {
-    uint32_t out;
-    uint32_t ret_val = _mulx_u32(a, b, &out);
-    return (out == 0) ? ret_val : UINT32_MAX;
+    if (aws_cpu_has_feature(AWS_CPU_FEATURE_BMI2)) {
+        uint32_t out;
+        uint32_t ret_val = _mulx_u32(a, b, &out);
+        return (out == 0) ? ret_val : UINT32_MAX;
+    } else {
+        /* If BMI2 unavailable, use __emulu instead */
+        uint64_t result = __emulu(a, b);
```

Comment:
is it guaranteed to be available everywhere?

Reply:
seems like its base x86 intrinsic judging by https://learn.microsoft.com/en-us/cpp/intrinsics/x64-amd64-intrinsics-list?view=msvc-160

Reply:
why intel only? it should be available on all processors that support x84 or x64 instruction sets